ftoolz/functoolz/iter.py

```diff
@@ -156,6 +156,26 @@ def fproduct(
     return fmap(lambda a: (a, f(a)), fa)
 
 
+def generate(f: Callable[[int], A_out], n: int) -> Iterable[A_out]:
+    """
+    Create an iterable of `n` elements generated by `f``.
+
+    >>> def f(i: int) -> float:
+    ...     return float(i) ** 2
+
+    >>> list(generate(f, 4))
+    [0.0, 1.0, 4.0, 9.0]
+
+    >>> list(generate(f, 0))
+    []
+
+    >>> list(generate(f, -1))
+    []
+    """
+    for i in range(n):
```

Comment:
one could add the optional parameters as they are given for range, not just n.

Reply:
good idea, will update it

Reply:
updated and passing tests, check it out ;)